File: Data Structures & Algorithms/validate-parentheses/submission-0.js
class Solution {
    /**
     * @param {string} s
     * @return {boolean}
     */
    isValid(s) {
        let stack=[]
        for(let br of s){
            if(br=="("||br=="{"||br=="["){
                stack.push(br)
            }
            else if(br==")"&&stack[stack.length-1]=="("){
                stack.pop()
            }
            else if(br=="}"&&stack[stack.length-1]=="{"){
                stack.pop()
            }
            else if(br==="]"&&stack[stack.length-1]=="["){
                stack.pop()
            }
            else{
                return false
            }
        }
    return stack.length==0
    }
}

File: Data Structures & Algorithms/combinations-of-a-phone-number/submission-0.py
class Solution:
    def letterCombinations(self, digits: str) -> List[str]:
        if len(digits)==0:
            return []
        res=[]
        sub=[]
        dic={"2":"abc","3":"def","4":"ghi","5":"jkl","6":"mno","7":"pqrs","8":"tuv","9":"wxyz"}
        def dfs(i):
            if i==len(digits):
                res.append("".join(sub[:]))
                return
            st=dic[digits[i]]
            for char in st:
                sub.append(char)
                dfs(i+1)
                sub.pop()
            
        dfs(0)
        return res